File: Recursion/Palindrone/palin.cpp
#include <string>
using namespace std;

class Solution {
  public:
    bool checkPalindrome(string s, int start, int end){
       if(start >= end)return true;
       if(s[start]!=s[end])return false;
       return checkPalindrome(s,start+1,end-1);
    }
    bool isPalindrome(int n) {
        // code here.
        string s = to_string(n);
        return checkPalindrome(s,0,s.length()-1);
    }
};

File: Recursion/Palindrone/palin.py
class Solution:
    def checkPalindrome(self, s, start, end):
        if start >= end:
            return True
        if s[start] != s[end]:
            return False
        return self.checkPalindrome(s, start + 1, end - 1)

    def isPalindrome(self, n):
        s = str(n)
        return self.checkPalindrome(s, 0, len(s) - 1)
